Commonality Analysis Code.Rmd

```diff
@@ -0,0 +1,258 @@
+---
+title: "COPE Commonality Analysis"
+author: "Arielle Smith and Laura Jans"
+date: "2023-06-30"
+output: html_document
+---
+
+```{r setup}
+#Load required packages
+if(!require(tidyverse)){install.packages('tidyverse')}
+library(tidyverse)
+if(!require(missForest)){install.packages('missForest')}
+library(missForest)
+if(!require(corrplot)){install.packages('corrplot')}
+library(corrplot)
+if(!require(naniar)){install.packages('naniar')}
+library(naniar)
+if(!require(forcats)){install.packages('forcats')}
+library(forcats)
+if(!require(psych)){install.packages('psych')}
+library(psych)
+
+#Set working directory
+#setwd("/Volumes/jslab/Commonality Analysis") #Arielle
+setwd("Z:/Commonality Analysis") #Laura
+
+#Read in data
+cope_full_data <- readRDS("cleaned_cope_data_randomized.rds")
+```
+
+```{r data cleaning}
+#Select necessary columns
+cope_ca_data <- cope_full_data %>% 
+  select(contains(c("b_dem", "b_bhs", "b_shs", "b_cdi", "b_gad", "b_cts_rs","b_drs_month", "b_pfs", "pi_bhs", "pi_shs", "f1_bhs", "f1_shs", "f1_cdi", "f1_gad", "f1_cts_rs", "f1_drs_month", "condition"))) %>%
+#Adding cdi sums (means already exist for all the other variables - which is what we need for our analyses!)
+  mutate(b_cdi_sum = b_cdi_1 + b_cdi_2 + b_cdi_3 + b_cdi_4 + b_cdi_5 + b_cdi_6 + b_cdi_7 + b_cdi_8 + b_cdi_9 + b_cdi_10 + b_cdi_11 + b_cdi_12, 
+         f1_cdi_sum = f1_cdi_1 + f1_cdi_2 + f1_cdi_3 + f1_cdi_4 + f1_cdi_5 + f1_cdi_6 + f1_cdi_7 + f1_cdi_8 + f1_cdi_9 + f1_cdi_10 + f1_cdi_11 + f1_cdi_12) %>% 
+#Confusing that the program feedback scale was labelled with the abbreviation for baseline rather than post-intervention, so changing it
+  rename_with(~ gsub("^b_pfs", "pi_pfs", .), starts_with("b_pfs"))
+
+#Includes the demographics, predictor variables, outcome variables, and the condition. Condition is just labeled 0, 1, and 2, but looking at the COPE baseline survey flow and parent study code, have gathered that:
+#0 = control (Sharing Feelings)
+#1 = intervention (Project Personality)
+#2 = intervention (ABC Project)
+
+#Make a new variable to indicate SSI completion - same method as code from parent project:
+cope_ca_data <- cope_ca_data %>% 

```

Comment:
This ended up producing an error message strangely, so I left it as is. Doesn't mutate make the pi_dropout object?

Error in `mutate()`:
! Problem while computing `..1 = pi_dropout == is.na(pi_pfs_1)`.
Caused by error in `mask$eval_all_mutate()`:
! object 'pi_dropout' not found

Reply:
Yes it should. This is an odd error message and might indicate that R is running a different `mutate()` function (i.e., one from another package). You can specify the package by doing `dplyr::mutate()` instead of `mutate()` or by using the `conflict_prefer()` function in the "conflicted" package (https://www.tidyverse.org/blog/2018/06/conflicted/).